spec/system/admin/reports/revenues_by_hub_spec.rb

```diff
@@ -153,13 +137,21 @@
         "20170",
         "Victoria",
         "1",
-        # 160.0$ - 9.5$
-        150.5,
-        # 10$ tax - 0.5$ voucher_excluded_tax
-        9.5,
-        # 5 line_items at 10$ each + 1 line_item at 110$
-        160.0
+        160.64,
+        10.36,
+        171.0
       ].compact.join(" "))
     end
   end
+
+  def apply_voucher(order, voucher)
+    voucher.create_adjustment(voucher.code, order)
+
+    # Update taxes
+    order.create_tax_charge!
+    order.update_shipping_fees!
+    order.update_order!
+
+    VoucherAdjustmentsService.new(order).update
```

Comment:
You need to add `order.update_totals_and_states` after `VoucherAdjustmentsService.new(order).update` otherwise the various order total won't take into account the voucher adjustment you just updated.

Reply:
thanks, I updated the specs as you suggested